mindmeld/active_learning/results_manager.py

```diff
@@ -0,0 +1,190 @@
+import os
+import json
+import datetime
+import logging
+from typing import Dict, List
+
+from mindmeld.markup import dump_query
+from ..path import (
+    AL_PARAMS_PATH,
+    AL_RESULTS_FOLDER,
+    AL_PLOTS_FOLDER,
+    AL_ACCURACIES_PATH,
+    AL_SELECTED_QUERIES_PATH,
+)
+from ..constants import STRATEGY_ABRIDGED
+
+logger = logging.getLogger(__name__)
+
+
+def create_dir_if_absent(base_path: str):
+    """Create a directory if one doesn't already exist at the given path.
+    Args:
+        base_path (str): Path to create directory."""
+    try:
+        os.makedirs(base_path)
+    except FileExistsError:
+        pass
+
+
+class ResultsManager:
+    """Handles the initialization of generated folder and its contents. Keeps record of experiment
+    results."""
+
+    def __init__(
+        self,
+        output_folder: str,
+    ):
+        """
+        Args:
+            output_folder (str): Directory to create an experiment folder or save log queries.
+        """
+        self.output_folder = output_folder
+        self.experiment_folder_name = None
+
+    def set_experiment_folder_name(self, selection_strategies) -> str:
+        """
+        Args:
+            selection_strategies (list): List of strategies used for the experiment.
+        Returns:
+            experiment_folder_name (str): Creates the name of the current experiment folder
+                based on the current timestamp.
+        """
+        strategies = "_".join(
+            STRATEGY_ABRIDGED[s] for s in selection_strategies if s in STRATEGY_ABRIDGED
+        )
+        now = datetime.datetime.now()
+        self.experiment_folder_name = (
+            f"{now.month}-{now.day}_{now.hour}:{now.minute}_{strategies}"
```

Comment:
maybe add year as well?

Reply:
Also why `{now.month}-{now.day}` but `{now.day}_{now.hour}`?

Reply:
Wanted to keep separation between date and time visually:

The following generates `3-12_3:42` which is easier to read than if it were all dashes.

Reply:
Added year